tests/test_pyrflx.py

```diff
@@ -1557,8 +1556,73 @@ def test_checksum_icmp(icmp_checksum: MessageValue) -> None:
+        b"\x4a\xfc\x0d\x00\x00\x00\x00\x00\x10\x11\x12\x13\x14\x15\x16\x17"
+        b"\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f\x20\x21\x22\x23\x24\x25\x26\x27"
+        b"\x28\x29\x2a\x2b\x2c\x2d\x2e\x2f\x30\x31\x32\x33\x34\x35\x36\x37"
+    )
+    icmp_checksum.set_checksum_function({"Checksum": icmp_checksum_function})
+    icmp_checksum.set("Tag", "Echo_Request")
+    icmp_checksum.set("Code_Zero", 0)
+    icmp_checksum.set("Identifier", 5)
+    icmp_checksum.set("Sequence_Number", 1)
+    icmp_checksum.set("Data", test_data)
+    assert icmp_checksum.get("Checksum") == 12824
+    assert icmp_checksum.bytestring == b"\x08\x00\x32\x18\x00\x05\x00\x01" + test_data
+    assert icmp_checksum.valid_message
+
+
+def test_checksum_auto_icmp_change_field(icmp_checksum: MessageValue) -> None:
+    test_data = (
+        b"\x47\xb4\x67\x5e\x00\x00\x00\x00"
+        b"\x4a\xfc\x0d\x00\x00\x00\x00\x00\x10\x11\x12\x13\x14\x15\x16\x17"
+        b"\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f\x20\x21\x22\x23\x24\x25\x26\x27"
+        b"\x28\x29\x2a\x2b\x2c\x2d\x2e\x2f\x30\x31\x32\x33\x34\x35\x36\x37"
+    )
+    icmp_checksum.set_checksum_function({"Checksum": icmp_checksum_function})
+    icmp_checksum.set("Tag", "Echo_Request")
+    icmp_checksum.set("Code_Zero", 0)
+    icmp_checksum.set("Identifier", 5)
+    icmp_checksum.set("Sequence_Number", 1)
+    icmp_checksum.set("Data", test_data)
     assert icmp_checksum.get("Checksum") == 12824
     assert icmp_checksum.bytestring == b"\x08\x00\x32\x18\x00\x05\x00\x01" + test_data
+    assert icmp_checksum.valid_message
+    icmp_checksum.set("Sequence_Number", 2)
+    icmp_checksum.set("Data", test_data)
+    assert icmp_checksum.get("Checksum") == 12823
+    assert icmp_checksum.bytestring == b"\x08\x00\x32\x17\x00\x05\x00\x02" + test_data
+    assert icmp_checksum.valid_message
+
+
+def test_checksum_parse_icmp(icmp_checksum: MessageValue) -> None:
+    test_data = (
+        b"\x08\x00\x32\x18\x00\x05\x00\x01\x47\xb4\x67\x5e\x00\x00\x00\x00"
+        b"\x4a\xfc\x0d\x00\x00\x00\x00\x00\x10\x11\x12\x13\x14\x15\x16\x17"
+        b"\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f\x20\x21\x22\x23\x24\x25\x26\x27"
+        b"\x28\x29\x2a\x2b\x2c\x2d\x2e\x2f\x30\x31\x32\x33\x34\x35\x36\x37"
+    )
+    icmp_checksum.set_checksum_function({"Checksum": icmp_checksum_function})
+    icmp_checksum.parse(test_data)
+    assert icmp_checksum.valid_message
+
+
+def test_invalid_checksum_parse_icmp(icmp_checksum: MessageValue) -> None:
+    test_data = (
+        b"\x08\x00\x35\x19\x00\x05\x00\x01\x47\xb4\x67\x5e\x00\x00\x00\x00"
+        b"\x4a\xfc\x0d\x00\x00\x00\x00\x00\x10\x11\x12\x13\x14\x15\x16\x17"
+        b"\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f\x20\x21\x22\x23\x24\x25\x26\x27"
+        b"\x28\x29\x2a\x2b\x2c\x2d\x2e\x2f\x30\x31\x32\x33\x34\x35\x36\x37"
+    )
+    icmp_checksum.set_checksum_function({"Checksum": icmp_checksum_function})
+    icmp_checksum.parse(test_data)
+    assert not icmp_checksum.valid_message
```

Comment:
We should also test that an exception is raise when trying to access the data (here, or in a separate test)

Reply:
I would move this to #423.

Reply:
Fine with me.

Reply:
I would recommend adapting the description of #423 accordingly.

Reply:
Done.